### ISSUE TITLE
how to visualise attention masks?

### ISSUE THREAD

**Issue by abucka** [tensorflow/models#10131]
Hello,

I trained an AOCR model and exported it to SavedModel format. When I run model cli command to check the inputs and outputs of the model, I see the attention masks for each char. The attention mask has a shape of (1,484) which indicates a vector. Using this information, how can I visualise the attentions in the input image?

**Comment by ymodak** [comment]
Have you tried https://github.com/tensorflow/tensor2tensor/tree/master/tensor2tensor/visualization ?

**Comment by abucka** [comment]
No I haven't, could you give me a hint how could I use it to process the SavedModel format output of aocr?

**Comment by ymodak** [comment]
> Have you tried https://github.com/tensorflow/tensor2tensor/tree/master/tensor2tensor/visualization ?

The above method requires your model to be saved in ckpt format as per the instructions.

For your case, You may want to try https://github.com/jessevig/bertviz 
They have clear steps to perform visualizations with their interactive colab.

**Comment by abucka** [comment]
But this transformers API applies to NLP and I am interested in Attention OCR: https://github.com/tensorflow/models/tree/master/research/attention_ocr. Using python/model_export.py script I get the SavedModel with the following outputs: attention_mask_0...n (n - number of chars detected). Each of these outputs has a shape of a vector (e.g. in my case (1,484)). I am not sure how to transform it to get the attention masks that can be visualised with input images. Do you have any hints?
